src/main/java/omero/cmd/graphs/BaseGraphTraversalProcessor.java

```diff
@@ -38,8 +49,14 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(BaseGraphTraversalProcessor.class);
 
+    /* compare with ome.services.delete.files.FileDeleter.Type */
+    private static final Collection<String> DELETION_CLASS_NAMES = ImmutableSet.of(
+            OriginalFile.class.getName(), Pixels.class.getName(), Thumbnail.class.getName());
+
     protected final Session session;
 
+    private final List<Map.Entry<String, Collection<Long>>> deletionLog = new ArrayList<>();
```

Comment:
Would a `ListMultimap` not serve as a better stoage solution?

Reply:
You're thinking a `List<ListMultimap<String, Long>>`? (To preserve the ordering of the deletion batches.) I figured that might need to somehow unpack and repack all the longs, whereas this way they just stay in the collection they're already in.

Reply:
I was thinking `ArrayListMultimap` on it's own, but I'm assuming that all keys are unique.

Reply:
Ah, no, I'm afraid we may get a bunch of `OriginalFile:3,5,7` then subsequently `OriginalFile:2,4,8` then have to remember to delete the first batch before the second.